### ISSUE TITLE
2.3.4 - 'RangeError [ERR_UNKNOWN_MODULE_FORMAT]: Unknown module format: undefined'

### ISSUE THREAD

**Issue by koshic** [iambumblehead/esmock#222]
After update from 2.3.3, I have a few errors in one of my projects, so obviously it related with
https://github.com/iambumblehead/esmock/blob/35ada4d7c8f6b4d21887eca1387f27bd756eb433/src/esmockLoader.js#L149
and change from hardcoded 'module' to 'context.format' introrduced in https://github.com/iambumblehead/esmock/pull/220.

Short analysis (4am, so...) - we modify source for esmock-specific module url, so who are responsible to set proper format previously, inside 'resolve' hook? Right, esmock. But I can't find any reference to 'format' inside resolve hook:
https://github.com/iambumblehead/esmock/blob/35ada4d7c8f6b4d21887eca1387f27bd756eb433/src/esmockLoader.js#L69-L119
It's missed, and probably should be a part of https://github.com/iambumblehead/esmock/pull/220.

**Comment by tommy-mitchell** [comment]
Maybe `module` could be used as a fallback? e.g.

```js
format: context.format ?? 'module',
```

**Comment by koshic** [comment]
> Maybe `module` could be used as a fallback? e.g.
> 
> ```js
> format: context.format ?? 'module',
> ```

Sure, it solves the issue - but not the root cause. Why do we expect anything in 'context.format'? Other loaders / Node default loader may resolve module format, may not. Only esmock has knowledge about files with urls like 'file:///Users/xxxxxxxx/packages/create-request/src/create-request.ts?esmk=1'

**Comment by iambumblehead** [comment]
oh no I'm sorry everyone

**Comment by iambumblehead** [comment]
npm won't let me unpublish it

**Comment by iambumblehead** [comment]
Let's use this. If problems are still there for we could completely revert things back to 2.3.3 state and publish as 2.4.0
```javascript
format: context.format === 'commonjs' ? 'commonjs' : 'module',
```

**Comment by iambumblehead** [comment]
esmock@2.3.5 is published

**Comment by iambumblehead** [comment]
@koshic I agree with your comments and am interested to try removing "format" altogether. I don't remember why it is used and it could be something that was used for node 12 or node 14 which esmock did support but there was a problem with npm that forced us to drop node 14 from the test matrix

**Comment by iambumblehead** [comment]
for now, 2.3.5 is closer to the original 2.3.3 but still gives the 'commonjs' needed for the issue resolved by 2.3.4
```javascript
// 2.3.3
format: 'module',
// 2.3.4
format: context.format,
// 2.3.5
format: context.format === 'commonjs' ? 'commonjs' : 'module',
```
